### DISCUSSION TITLE
bad CPU type in executable: gs

### DISCUSSION THREAD

**Discussion by digitrance** [Homebrew/discussions#2047] [Everyday usage]
I'm trying to use ghostscript on a Mac running macOS Big Sur 11.5.2 on an Intel Core i5 CPU. I keep getting the error message: "bad CPU type in executable: gs".

I suspect it's because the ghostscript executable is 32-bit, which is no longer compatible with the latest versions of macOS.

% which gs
/usr/local/bin/gs

% ls -la /usr/local/bin/gs
lrwxr-xr-x  1 ____  ____  35 Aug 27 23:09 /usr/local/bin/gs -> ../Cellar/ghostscript/9.54.0/bin/gs

% file /usr/local/bin/gs
/usr/local/bin/gs: Mach-O universal binary with 2 architectures: [i386:Mach-O executable i386] [ppc]
/usr/local/bin/gs (for architecture i386):	Mach-O executable i386
/usr/local/bin/gs (for architecture ppc):	Mach-O executable ppc

Not sure whether there is something I can do on my machine to fix this, and/or whether the problem ultimately needs to be addressed further upstream.

**Comment by carlocab** [comment]
The `gs` executable shipped with Ghostscript is a native binary:
```
❯ file /usr/local/bin/gs
/usr/local/bin/gs: Mach-O 64-bit executable x86_64
```
What's the output of
```
brew config
brew doctor
brew info ghostscript
```
?

**Reply by digitrance** [reply]
carlocab,

Thansk for the pointers! I've included the output below, but ultimately was able to resolve the issue by doing 'brew uninstall --ignore-dependencies ghostscript' and 'brew install ghostscript'. In retrospect, maybe I should have tried that first, but I wasn't 100% sure whether the Homebrew-supplied binary was supposed to be x64 compatible, and I've never run into any real problems with Homebrew before.

I ran the commands you suggested before the uninstall/reinstall of ghostscript, and have provided the output below:

----

% brew config
HOMEBREW_VERSION: 3.2.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: d14fc2db125c18832fe83c98fdeb54931dc48f03
Last commit: 7 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 29cec7fbe03f9e154bee16eb1058ccef9a61ece8
Core tap last commit: 9 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit broadwell
Clang: 12.0.5 build 1205
Git: 2.30.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.5.2-x86_64
CLT: 12.5.0.22.9
Xcode: 12.5.1

% brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libssl.0.9.8.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/ulockmgr.h

% brew info ghostscript
ghostscript: stable 9.54.0 (bottled), HEAD
Interpreter for PostScript and PDF
https://www.ghostscript.com/
/usr/local/Cellar/ghostscript/9.54.0 (683 files, 149.0MB) *
  Poured from bottle on 2021-05-24 at 18:11:27
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/ghostscript.rb
License: AGPL-3.0-or-later
==> Dependencies
Build: pkg-config ✔
Required: fontconfig ✔, freetype ✔, jbig2dec ✔, jpeg ✔, libidn ✔, libpng ✔, libtiff ✔, little-cms2 ✔, openjpeg ✔
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 59,209 (30 days), 221,928 (90 days), 1,065,214 (365 days)
install-on-request: 10,458 (30 days), 39,952 (90 days), 180,227 (365 days)
build-error: 0 (30 days)

----

**Reply by carlocab** [reply]
All binary executables shipped by Homebrew should be natively compiled to work on your system. If one isn't, something's gone wrong.

Nothing in the info you've provided seems to suggest an explanation, unfortunately, but I guess we don't need one anymore. Glad you're sorted.
